### PR TITLE
feat: include service and definition types with implementations

### PR DESCRIPTION
**Pull request by niedzielski**
When exporting a service or service definition, also export it as a
type. Previously, you would have to import the value then `typeof` off
of it which is cumbersome. Automate this as it has no runtime overhead
and seems more intuitive.

Before:

```ts
export const TestDefinition = {
  name: 'Test',
  fullName: 'simple.Test',
  methods: {
    …
  },
} as const;
```

After:

```ts
export type TestDefinition = typeof TestDefinition;
export const TestDefinition = {
  name: 'Test',
  fullName: 'simple.Test',
  methods: {
    …
  },
} as const;
```

### CODE REVIEW

**Comment by fizx** [comment]
Hi @stephenh, 
Code looks good here, and it's a useful feature, but I was holding off on merging it because it's from someone on my work team, and we haven't chatted yet about whether you feel that represents a conflict of interest on my review.

Thanks!

**Review by fizx — COMMENTED** [review]
:shipit:

**Review by stephenh — APPROVED** [review]
Nice! Thanks @niedzielski !

Sorry for the really late review here; have just been busy. Fwiw @fizx yeah I'm perfectly happy with you + other maintainers (cc @boukeversteegh who's been actively lately as well) reviewing and merging PRs, to keep me from being the bottleneck on the project.

**Pull request merged**
